src/hf_hydrodata/point.py

```diff
@@ -1986,37 +1962,38 @@ def _get_data_nc(site_list, var_id, *args, **kwargs):
     else:
         options = kwargs
 
-    dirpath = _get_dirpath(var_id)
-    file_list = [f"{dirpath}/{site}.nc" for site in site_list]
-
-    varname_map = {
-        "1": "streamflow",
-        "2": "streamflow",
-        "3": "wtd",
-        "4": "wtd",
-        "5": "wtd",
-        "6": "swe",
-        "7": "precip_acc",
-        "8": "precip_inc",
-        "9": "precip_inc_sa",
-        "10": "temp_min",
-        "11": "temp_max",
-        "12": "temp_avg",
-        "13": "sms_2in",
-        "14": "sms_4in",
-        "15": "sms_8in",
-        "16": "sms_20in",
-        "17": "sms_40in",
-        "18": "latent heat flux",
-        "19": "sensible heat flux",
-        "20": "shortwave radiation",
-        "21": "longwave radiation",
-        "22": "vapor pressure deficit",
-        "23": "air temperature",
-        "24": "wind speed",
-    }
+    # Users input a `depth_level` parameter to filter different soil moisture variables
+    # This is only distinguished in the data catalog with the dataset_var field, so this
+    # provides a mapping between the two to translate these queries.
+    sm_vars = {2: "sms_2in", 4: "sms_4in", 8: "sms_8in", 20: "sms_20in", 40: "sms_40in"}
+
+    # For soil moisture queries, we need the additional dataset_var filter to make the
+    # request un-ambiguous.
+    if "depth_level" in options and options["depth_level"] is not None:
```

Comment:
Is depth level only needed for soil moisture, not for some of the geo-fabric variables (maybe that logic is in `gridded.py`)? It almost seems like this logic could live in the `get_catalog_entry` function, just a thought.
Either way, I wonder what happens if a user has a depth level and the variable isn't soil moisture or a depth level not in the `sm_vars` dictionary. There's a lot of code here so maybe this is already handled, just want to make sure that edge case is handled somewhere.

Reply:
@welytle Great question. If this becomes more prevalent, I think that would be a good reason to think about how to put this in a more general function like `get_catalog_entry`. Right now, soil moisture is the only point variable that has multiple depth levels and there is some logic in a `_check_inputs` utility function within the point module to ensure the user has provided a valid depth level value if they are requesting soil moisture data. I have tests that check if a user is asking for soil moisture and either 1.) does not provide a depth level or 2.) provides an unsupported depth level. But good catch on the other case: I'm not handling explicitly what happens if a user adds a depth level but is _not_ requesting soil moisture data. I added that check in along with an additional test. I also created a variable to make it clear from the top what accepted depth options are.

For context, the gridded data files that have a z/depth dimension always return the full depth grid for a user, so it's not a user option over there to extract a single depth level if there are multiple available. For the point data, we only support returning 2D data at the moment (single time series for potentially multiple sites) so we make a user choose the depth level.

Reply:
Thanks for all the context!